flutter/lib/src/default_integrations.dart

```diff
@@ -195,14 +196,87 @@ class _LoadContextsIntegrationEventProcessor extends EventProcessor {
         event = event.copyWith(contexts: eventContexts);
       }
 
+      final tagsMap = infos['tags'];
+      if (tagsMap != null) {
+        final tags = event.tags ?? {};
+        final newTags = Map<String, String>.from(tagsMap as Map);
+
+        for (final tag in newTags.entries) {
+          if (!tags.containsKey(tag.key)) {
+            tags[tag.key] = tag.value;
+          }
+        }
+        event = event.copyWith(tags: tags);
+      }
+
+      final extraMap = infos['extra'];
+      if (extraMap != null) {
+        final extras = event.extra ?? {};
+        final newExtras = Map<String, dynamic>.from(extraMap as Map);
+
+        for (final extra in newExtras.entries) {
+          if (!extras.containsKey(extra.key)) {
+            extras[extra.key] = extra.value;
+          }
+        }
+        event = event.copyWith(extra: extras);
+      }
+
       final userMap = infos['user'];
       if (event.user == null && userMap != null) {
         final user = Map<String, dynamic>.from(userMap as Map);
         event = event.copyWith(user: SentryUser.fromJson(user));
       }
 
-      if (infos['integrations'] != null) {
-        final integrations = List<String>.from(infos['integrations'] as List);
+      final distString = infos['dist'];
+      if (event.dist == null && distString != null) {
```

Comment:
```suggestion
      if (event.dist == null && distString?.isNotEmpty ?? false) {
```

Same for other fields